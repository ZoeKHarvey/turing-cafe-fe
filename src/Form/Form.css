form {
  display: flex;
  align-items: center
}

input {
  height: 5vh;
  font-size: 1.1em;
  margin-right: 2%;
}