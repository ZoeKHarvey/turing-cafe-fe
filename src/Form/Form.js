import React, { Component } from 'react';

export default class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      number: null,
      id: null,
    }
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value})
    console.log('handle change firing', this.state)
  }

  submitNewReservation = e => {
    e.preventDefault();
    const { addReservation } = this.props;
    const newReservation = { ...this.state, id: Date.now()};
    addReservation(newReservation);
    this.resetInputs();
  }

  resetInputs = () => {
    this.setState({ 
      name: '',
      date: '',
      time: '',
      number: null,
      id: null,
    })
  }

  render() {
    return (
      <form>
        <input
          type='text'
          placeholder='Name'
          value={this.state.name}
          name='name'
          onChange={this.handleChange}
        />
        <input
          type='date'
          placeholder='Date'
          value={this.state.date}
          name='date'
          onChange={this.handleChange}
        />
        <input
          type='time'
          placeholder='Time'
          value={this.state.time}
          name='time'
          onChange={this.handleChange}
        />
        <input
          type='number'
          placeholder='Number of Guests'
          value={this.state.number}
          name='number'
          onChange={this.handleChange}
        />
        <button 
          onClick={this.submitNewReservation}
        >
          Submit!
        </button>
      </form>
    )
  }
}
