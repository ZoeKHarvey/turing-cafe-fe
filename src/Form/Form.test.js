import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Form addReservation={jest.fn()} />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is called', () => {
    const mockEvent = { target: 
      { name: 'name', 
      value: 'DOG' } };
    const expected = 'DOG';

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state('name')).toEqual(expected);
  });

  it('should reset state when resetInputs is called', () => {
    const expected = { name: '', date: '', time: '', number: null };

    wrapper.instance().setState({ 
      name: 'doggo', date: '2020/05/02', time: '8:00', number: 3
    });
    
    wrapper.instance().resetInputs();

    expect(wrapper.state()).toEqual(expected);
  });

  it('should call resetInputs when submitNewReservation is called', () => {
    wrapper.instance().resetInputs = jest.fn();
    wrapper.instance().submitNewReservation({ preventDefault: jest.fn() });

    expect(wrapper.instance().resetInputs).toHaveBeenCalled();
  });

  it('should run submitReservation when the button is clicked', () => {
    wrapper.instance().submitNewReservation = jest.fn();
    wrapper.instance().forceUpdate();
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.find('button').simulate('click', mockEvent);

    expect(wrapper.instance().submitNewReservation).toHaveBeenCalled();
  });
});