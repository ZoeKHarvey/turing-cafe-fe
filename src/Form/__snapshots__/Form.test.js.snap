// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date"
    type="date"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="time"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="number"
    value={null}
  />
  <button
    onClick={[Function]}
  >
    Submit!
  </button>
</form>
`;
